### PR TITLE
[Feature] BE support IPv6

### PR DESCRIPTION
**Pull request by Lrabbit125**
## What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/src/service/backend_options.h [hunk 1]
```diff
@@ -33,7 +33,10 @@ class BackendOptions {
     static bool init();
     static std::string get_localhost();
     static TBackend get_localBackend();
+    static const char* get_service_bind_address();
+    static const char* get_service_bind_address_without_bracket();
```
Comment (imay) [inline be/src/service/backend_options.h:37]: how about returning std::string to avoid accessing memory issue?

File: be/src/util/network_util.cpp [hunk 2]
```diff
@@ -80,29 +80,89 @@ Status get_hostname(std::string* hostname) {
...
+Status get_hosts(std::vector<InetAddress>* hosts) {
+    ifaddrs* if_addrs = nullptr;
+    if (getifaddrs(&if_addrs)) {
+        std::stringstream ss;
+        char buf[64];
+        ss << "getifaddrs failed because " << strerror_r(errno, buf, sizeof(buf));
+        return Status::InternalError(ss.str());
+    }
+
+    for(ifaddrs* if_addr = if_addrs; if_addr != nullptr; if_addr = if_addr->ifa_next) {
```
Comment (imay) [inline be/src/util/network_util.cpp:92]:
```suggestion
    for (ifaddrs* if_addr = if_addrs; if_addr != nullptr; if_addr = if_addr->ifa_next) {
```

File: be/src/util/network_util.cpp [hunk 3]
```diff
@@ -80,29 +80,89 @@ Status get_hostname(std::string* hostname) {
...
+        if(addr->sa_family == AF_INET) {
+            //check legitimacy of IPv4 Addresses
+            char addr_buf[INET_ADDRSTRLEN];
+            auto tmp_addr = &((struct sockaddr_in*)if_addr->ifa_addr)->sin_addr;
+            inet_ntop(AF_INET, tmp_addr, addr_buf, INET_ADDRSTRLEN);
+            //check is loopback Addresses
+            in_addr_t s_addr = ((struct sockaddr_in*)if_addr->ifa_addr)->sin_addr.s_addr;
+            bool is_loopback = (ntohl(s_addr) & 0xFF000000) == 0x7F000000;
+            hosts->emplace_back(std::string(addr_buf), AF_INET, is_loopback);
+        }else if (addr->sa_family == AF_INET6) {
```
Comment (imay) [inline be/src/util/network_util.cpp:106]:
```suggestion
        } else if (addr->sa_family == AF_INET6) {
```

**Comment by Astralidea** [comment]
你可以运行build-support下面的clang-format.sh来进行格式化。另外可以增加一些单测提高覆盖率。

**Review by nshangyiming — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Pull request merged**